### DISCUSSION TITLE
Standalone app

### DISCUSSION THREAD

**Discussion by Reasonant** [AIrjen/OneButtonPrompt#127] [General]
Hi, I want to ask. Could you possibly make this tool as a standalone application for desktop ? I you cannot, would it be possible to tell me how to do it ? I really love this tool, but I don't want to open automatic1111 all the time. I want to have my prompts to a file. I also want to add more concepts, themes, subjects to the csv files, or create more templates.
You did great work and congratulations. If you could please answer this, because I would really like to have it as a standalone application. It does not need to be an .exe it could work with gradio. I just want the tool to work alone so I can add more things to it.
Thanks in advance.

(I managed to get the build_dynamic_prompt() function to work on Spyder. I also add more stuff to the csv files. But I cannot control the variables, for example to turn on or off specific art styles or subjects. I also tried to understand how the templates work but I didn't figure it out. My programming skills are limited even though I want to tweak this tool. Maybe you could help me.)

Also another thought : I thought of putting many prompts generated by OneButtonPrompt into ChatGPT and asking it to correct them to be more coherent and aesthetically pleasant. I didn't manage to do it as I want. Anyway, I suppose that we could feed many prompts into an LLM and then finetune it to produce better prompts. It is just an idea.

**Comment by AIrjen** [comment]
Hi! There are a lot of things that are already controllable. For your request:  _I also want to add more concepts, themes, subjects to the csv files, or create more templates._

Check here how to add things to csv's without breaking things on updates:
https://github.com/AIrjen/OneButtonPrompt/blob/main/user_guides/custom_files.md

Check here how to control certain parts of the generation process (such as randomness and subjects etc):
https://github.com/AIrjen/OneButtonPrompt/blob/main/user_guides/config_file.md

As for a standalone application. I am thinking about this, but I currently don't have a good development environment (some life things going on!). I might be able to do more in a couple of weeks. I do have another idea to make this available somewhere else on the web, but I need to find the time for it.

I thought about using LLM's, but I don't think it is worth it. The whole idea is to discover new and unexpected results. There are already prompt generators that use a model in the background. Check out Prompt Magic in dynamic prompts extension! https://github.com/adieyal/sd-dynamic-prompts#magic-prompt

**Comment by Reasonant** [comment]
Hey there, could you please tell me what I can change in the code so I can run the project as standalone ? Without the need to open Automatic111. Please.

**Reply by AIrjen** [reply]
Hi, you can take a look at prompttester.py. For testing purposes I use this file to run a bunch of prompts while I am developing.

In Python, all you need is :

from build_dynamic_prompt import *
prompt = build_dynamic_prompt()

Everything else is just setting the parameters up.

Hope that helps in giving you a headstart.

**Reply by AIrjen** [reply]
Hi, I finally got around to building a basic website that does prompt generation online. Check it out here: https://airjen.pythonanywhere.com/

![image](https://github.com/AIrjen/OneButtonPrompt/assets/130234949/51f17fc7-9c4e-4bc8-98b2-16bd3b011e4a)
